l2-contracts/contracts/ConsensusRegistry.sol

```diff
@@ -0,0 +1,478 @@
+    /// @dev Only callable by the contract owner.
+    /// @dev Verifies that the node owner exists in the registry.
+    /// @param _nodeOwner The address of the node's owner to be removed.
+    function remove(address _nodeOwner) external onlyOwner {
+        verifyNodeOwnerExists(_nodeOwner);
+
+        scheduleNodeRemoval(_nodeOwner);
+
+        emit NodeRemoved(_nodeOwner);
+    }
+
+    /// @notice Changes the validator weight of a node in the registry.
+    /// @dev Only callable by the contract owner.
+    /// @dev Verifies that the node owner exists in the registry.
+    /// @param _nodeOwner The address of the node's owner whose validator weight will be changed.
+    /// @param _weight The new validator weight to assign to the node.
+    function changeValidatorWeight(address _nodeOwner, uint32 _weight) external onlyOwner {
+        verifyNodeOwnerExists(_nodeOwner);
+
+        Node storage node = nodes[_nodeOwner];
+        ensureValidatorSnapshot(node);
+        node.validatorLatest.weight = _weight;
+
+        emit NodeValidatorWeightChanged(_nodeOwner, _weight);
+    }
+
+    /// @notice Changes the attester weight of a node in the registry.
+    /// @dev Only callable by the contract owner.
+    /// @dev Verifies that the node owner exists in the registry.
+    /// @param _nodeOwner The address of the node's owner whose attester weight will be changed.
+    /// @param _weight The new attester weight to assign to the node.
+    function changeAttesterWeight(address _nodeOwner, uint32 _weight) external onlyOwner {
+        verifyNodeOwnerExists(_nodeOwner);
+
+        Node storage node = nodes[_nodeOwner];
+        ensureAttesterSnapshot(node);
+        node.attesterLatest.weight = _weight;
+
+        emit NodeAttesterWeightChanged(_nodeOwner, _weight);
+    }
+
+    /// @notice Changes the validator's public key and proof-of-possession (PoP) in the registry.
+    /// @dev Only callable by the contract owner or the node owner.
+    /// @dev Verifies that the node owner exists in the registry.
+    /// @param _nodeOwner The address of the node's owner whose validator key and PoP will be changed.
+    /// @param _pubKey The new BLS12-381 public key to assign to the node's validator.
+    /// @param _pop The new proof-of-possession (PoP) to assign to the node's validator.
+    function changeValidatorKey(
+        address _nodeOwner,
+        BLS12_381PublicKey calldata _pubKey,
+        BLS12_381Signature calldata _pop
+    ) external onlyOwnerOrNodeOwner(_nodeOwner) {
+        verifyInputBLS12_381PublicKey(_pubKey);
+        verifyInputBLS12_381Signature(_pop);
+        verifyNodeOwnerExists(_nodeOwner);
+
+        Node storage node = nodes[_nodeOwner];
+        ensureValidatorSnapshot(node);
+        node.validatorLatest.pubKey = _pubKey;
+        node.validatorLatest.pop = _pop;
+
+        emit NodeValidatorPubKeyChanged(_nodeOwner, _pubKey);
+        emit NodeValidatorPoPChanged(_nodeOwner, _pop);
```

Comment:
Should this be one event? Ostensibly one can't change without the other.

Reply:
Fixed here: https://github.com/matter-labs/era-contracts/pull/555/commits/347229dcd8d8b12650d638af516d6232c42d9135